lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2574,46 +2552,92 @@ void ModuleEmitter::emitStatementBlock(Block &body) {
 }
 
 //===----------------------------------------------------------------------===//
-// Module Driver
+// SplitModuleEmitter
 //===----------------------------------------------------------------------===//
 
+namespace {
+
+class SplitModuleEmitter {
+public:
+  explicit SplitModuleEmitter(StringRef dirname)
+      : encounteredError(false), dirname(dirname) {}
+
+  /// Whether any error has been encountered during emission.
+  std::atomic<bool> encounteredError;
+
+  /// The directory to emit files into.
+  StringRef dirname;
+
+  /// A list of modules and their position within the per-file operations.
+  struct EmittedModule {
+    Operation *op;
+    size_t position;
+    SmallString<32> filename;
+  };
+  SmallVector<EmittedModule, 0> moduleOps;
+
+  /// A list of per-file operations (e.g., `sv.verbatim` or `sv.ifdef`).
+  SmallVector<Operation *, 0> perFileOps;
```

Comment:
Curiosity: why the `, 0` on these smallvectors?

Reply:
@lattner There was a line in the LLVM programmer's manual about using `, 0` to get `std::vector`-like behavior in case there is no obvious count that the vector generally stays below. Arguably for the `perFileOps` we could have something like 8 or 16. Not sure about the `moduleOps`, as these roughly scale linearly with input size.

Reply:
Got it, ok thanks!